### PR TITLE
Adapter: add mocha adapter

### PR DESCRIPTION
**Pull request by fcarstens**
This PR adds an adapter for Mocha. This is semi-blocked by #24, because the start event tests aren't on master yet.

### CODE REVIEW

**Comment by jzaefferer** [comment]
The only thing I'm missing is additions to `test/test.js`, but that's blocked by #24, right? Since that is good to land, let me know when you've updated this, and I'll take another look.

**Comment by fcarstens** [comment]
Added tests and fixed bugs. :smiley:

**Comment by fcarstens** [comment]
Seems like I didn't `git add` after linting. I'm on mobile now, will fix that tomorrow. :smirk:

**Comment by jzaefferer** [comment]
The next push should also build on Travis. While we've added the `.travis.yml` file, we didn't actually enable the build on Travis. That's now fixed.

File: lib/adapters/MochaAdapter.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    onFail(test, error) {
+        this.errors.push(error);
+    }
+
+    onTestEnd(details) {
+        var test = this.idDataMap[details.title];
+        if (details.pending) {
+            this.emit("testStart", test);
```
Comment (jzaefferer) [inline lib/adapters/MochaAdapter.js]: Wondering why this is needed - there is no `test` event for pending tests?

File: test/mocha/index.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="../../node_modules/mocha/mocha.css">
+    <script src="../../node_modules/mocha/mocha.js"></script>
+    <script>mocha.setup('bdd')</script>
```
Comment (jzaefferer) [inline test/mocha/index.html]: Does it make any difference for the adapter what argument is passed here?

File: test/mocha/tests.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    it("bar goes wrong", function () {
+        assert(5 === "5");
+    });
+});
+
+describe("group b", function () {
+    it("baz", function () {
+        assert(true);
+    });
+    xit("skipped test", function () {
```
Comment (jzaefferer) [inline test/mocha/tests.js]: Is there also `xdescribe`? Should check for that. Not quite sure what to make of this: https://github.com/mochajs/mocha/commit/6cdca19eafa34bc8f05cfa69a8084fdd6b2cf06f

**Comment by jzaefferer** [comment]
Mostly looking good. Let's fix this up and land it.

**Comment by fcarstens** [comment]
Thanks for reviewing! :smiley:
I have my last exam on the 30th of September and will update things afterwards. :relaxed:

**Comment by jzaefferer** [comment]
@fcarstens I'm available this week to do more review, let me know when you've updated this.

File: test/mocha/tests.js (hunk 3 above)
Comment (cynicaldevil) [inline test/mocha/tests.js]: Maybe its wiser to have the more conventional `.skip()` test too? `xdescribe()` and `xit()` maybe deprecated in the future.
http://tjholowaychuk.tumblr.com/post/29996505898/mocha-140

**Comment by jzaefferer** [comment]
You can use the code/branch from this PR as a starting point, and add commits on top. That way @fcarstens contribution can be kept intact, while we can move forward. There's no need to copy anything.

Let me know if you need help doing the above with git. This might be a good starting point, too: http://stackoverflow.com/a/14869954/2671

File: test/mocha/tests.js (hunk 3 above)
Comment (jzaefferer) [inline test/mocha/tests.js]: Yeah, makes sense.

**Comment by cynicaldevil** [comment]
So I have to add fcartens' repo as a remote , fetch it and then start working on the mocha_adapter branch?

**Comment by jzaefferer** [comment]
Yes on both!

**Comment by jzaefferer** [comment]
Replaced by #36

**Pull request closed without merging**